src/Helper/Product/Type/Configurable.php

```diff
@@ -70,7 +70,12 @@ public function getChildren()
         $productCollection->addAttributeToFilter('entity_id', ['in' => $childProductIds]);
         $productCollection->addStoreFilter();
 
-        return $productCollection;
+        $children = [];
+        foreach ($productCollection as $product) {
```

Comment:
so you optimized crappy code to make it faster and now made it worse than the original? :D
1) Seems like now you load a collection so that you can get childId's that you already have?
2) you use load method which is deprecated, the previous solution used a repository to do this
3) loading may not be needed, I am assuming you need some properties that are not loaded via collection call? You could add things to select additionally by using `$collections->addAttributesToSelect` method to add additional stuff, or use joins to make it more performant. Loading individually kinda does suck.